tagger.go

```diff
@@ -65,85 +69,240 @@ func newLinodeClient() linodego.Client {
+
+			instanceType := "volumes"
+			instanceTagMap[instanceType] = make(map[int][]string)
 
-		if len(combinedNewTags) > 0 {
-			linodeIDTagMap[linode.ID] = combinedNewTags
-			sort.Strings(combinedNewTags)
-			if !slices.Equal(tags, combinedNewTags) {
-				log.WithFields(log.Fields{
-					"linode_id": linode.ID,
-					"old_tags":  tags,
-					"new_tags":  combinedNewTags,
-				}).Debug("Linode tag set updated")
+			if linodeTagSlice[instanceType] != nil {
+				for _, volume := range objects {
+					combinedNewTags, tags, linodeID = compareTags(volume, linodeTagSlice[instanceType])
+					instanceTagMap[instanceType][linodeID] = combinedNewTags
+					logTags(combinedNewTags, tags, linodeID)
+				}
+			}
+		case []linodego.NodeBalancer:
+
+			instanceType := "nodebalancers"
+			instanceTagMap[instanceType] = make(map[int][]string)
+
+			if linodeTagSlice[instanceType] != nil {
+				for _, nodebalancer := range objects {
+					combinedNewTags, tags, linodeID = compareTags(nodebalancer, linodeTagSlice[instanceType])
+					instanceTagMap[instanceType][linodeID] = combinedNewTags
+					logTags(combinedNewTags, tags, linodeID)
+				}
+			}
+		case []linodego.Domain:
+
+			instanceType := "domains"
+			instanceTagMap[instanceType] = make(map[int][]string)
+
+			if linodeTagSlice[instanceType] != nil {
+				for _, domain := range objects {
+					combinedNewTags, tags, linodeID = compareTags(domain, linodeTagSlice[instanceType])
+					instanceTagMap[instanceType][linodeID] = combinedNewTags
+					logTags(combinedNewTags, tags, linodeID)
+				}
+			}
+		case []linodego.LKECluster:
+
+			instanceType := "lkeclusters"
+			instanceTagMap[instanceType] = make(map[int][]string)
+
+			if linodeTagSlice[instanceType] != nil {
+				for _, lkecluster := range objects {
+					combinedNewTags, tags, linodeID = compareTags(lkecluster, linodeTagSlice[instanceType])
+					instanceTagMap[instanceType][linodeID] = combinedNewTags
+					logTags(combinedNewTags, tags, linodeID)
+				}
 			}
 		}
+
 	}
 
-	return linodeIDTagMap, nil
+	return instanceTagMap, nil
 }
 
-func updateLinodeInstanceTags(ctx context.Context, client linodego.Client, id int, tags *[]string) error {
-	updatedInstance, err := client.UpdateInstance(ctx, id, linodego.InstanceUpdateOptions{Tags: tags})
-	if err != nil {
-		return err
-	}
+func updateLinodeInstanceTags(ctx context.Context, client linodego.Client, id int, tags *[]string, data string) error {
+
+	if id != 0 {
+		switch data {
+		case "linodes":
+			updatedInstance, err := client.UpdateInstance(ctx, id, linodego.InstanceUpdateOptions{Tags: tags})
+			if err != nil {
+				return err
+			}
+			sort.Strings(*tags)
+			updatedTags := updatedInstance.Tags
+			sort.Strings(updatedTags)
+
+			if !slices.Equal(updatedTags, *tags) {
+				return errors.New("call to update instance did not result in the expected tag set")
+			}
+		case "volumes":
+			updatedInstance, err := client.UpdateVolume(ctx, id, linodego.VolumeUpdateOptions{Tags: tags})
```

Comment:
```suggestion
		case "volumes":
			updatedVolume, err := client.UpdateVolume(ctx, id, linodego.VolumeUpdateOptions{Tags: tags})
```
Perhaps I'm being too nit picky here, but this is another section where I feel it might read a bit clearer with a new name. I don't really care what, it only exists in this scope anyway. `obj`, `v`, whatever. just not referring to each of these inner scoped updated objects as instances sounds like a bit clearer to me